tasty/skyspark/point_mapper.py

```diff
@@ -0,0 +1,148 @@
+from rdflib import OWL, RDF, RDFS
+from tasty import constants as tc
+from tasty import graphs as tg
+
+
+class PointNode:
+
+    # -- Class Constructor --
+    def __init__(self, type: str, parent):
+        self.type = type
+        self.parent = parent
+        self.children = []
+        self.tags = []
+        self.load_tags()
+
+    def add_child(self, child: 'PointNode'):
+        """
+        Add a child (node) to the given node
+
+        :param child: the child node to add
+        :type child: PointNode
+        """
+        self.children.append(child)
+
+    def load_tags(self):
+        """
+        Load the tags for this node based on the 'type' string passed in at instantiation. This method will split the
+        'type' on hyphens and add each word as a tag to the node. The tag "point" is removed (this may be updated at
+        a later time)
+        """
+        self.tags = self.type.split("-")
+
+        # remove empty strings and 'point'
+        # eventually we may want "point" as a tag, but it's currently not in the "source" or "generated" shapes
+        for tag in self.tags:
+            if(tag == '' or tag == 'point'):
+                self.tags.remove(tag)
+
+# Note this data structure implicitly assumes that points fall into a tree-like hierarchy and that each point can only be
+# a subtype of ONE point type. This may not be the actual case in some systems and may need to be re-considered at another
+# time. Also because of this, if 'his', 'cur', 'writable', or 'weather' tags are not removed, first class point may not be
+# correct, as it assumes mutual exclusivity.
+
+
+class PointTree:
+
+    # -- Class Constructor --
+    def __init__(self, filename: str, root_type: str):
+        self.root_type = root_type
+        self.filename = filename
+        self.root = None
+        self.generate_point_tree(filename)
+
+    def get_root(self) -> 'PointNode':
+        return self.root
+
+    def generate_point_tree(self, filename: str):
+        """
+        Generates/populates a PointTree based on a schema definition file (e.g. 'schemas/haystack/defs_3_9_10.ttl') which
+        describes the desired ontology.
+
+        :param filename: the filepath/filename from which to generate the PointTree
+        """
+        # parse the file to an rdf graph
+        graph = tg.get_versioned_graph(tc.HAYSTACK, tc.V3_9_10)
+        graph.parse(filename, format='turtle')
+
+        # get root of graph from the rdf graph
+        root_uri = tc.PHIOT_3_9_10[self.root_type]
```

Comment:
Yes it could in theory, as long as the schema has a tree-like hierarchy of point types. That being said, the brick schema definition files seem to have a slightly different structure, in terms of the use of `rdfs:subClassOf` that I don't fully understand at the moment and which would likely require some modification to the methods here.

Reply:
Also, if we do, it might be worth updating the ` PointNode.load_tags` method to take in a delimiting character as right now a hyphen is hard-coded.

Reply:
My comment was for Haystack since I think that's implied by including it in a SkySpark file.